### ISSUE TITLE
Incorrect calculation of MSE in LossCalculation.score()

### ISSUE THREAD

**Issue by danielr2e** [deeplearning4j/deeplearning4j#1942]
For the MSE case, the summed error is being multiplied by 0.5 instead of divided by the number of examples. Ultimately this results in MSE 'scores' that are very different from whatever logic is computing them in RegressionEvaluation.

**Comment by eraly** [comment]
@danielr2e So the scores get divided by the size of the minibatch and that is the number of examples. If you gave us some data that will help us reproduce the problem that would be great.

**Comment by anovstrup** [comment]
We're seeing this issue in the context of training/evaluating a recurrent neural network with truncated BPTT. In the training context, `exampleScores.rows() >= miniBatchSize * tBPTTLength` (it's an inequality because some time steps may be masked).

The solution may be to always divide by `mask.sumNumber().doubleValue()` instead of multiplying by half and dividing by miniBatchSize?

**Comment by eraly** [comment]
Oh that makes total sense. Thanks.

**Comment by AlexDBlack** [comment]
As for how we handle time series: scores are the sum of the scores for each time series. i.e., long time series have larger scores than short time series.

In this case, I think dividing by number of examples (number of time series) is still correct? The mask array is taken into account in the sum.

**Comment by AlexDBlack** [comment]
As for scores vs. regression evaluation: I'd have to take a look there. Is the difference due to sum vs. average over the entire test set?

**Comment by anovstrup** [comment]
We used `RegressionEvaluation.evalTimeSeries` to compute the MSE, which correctly computes the mean over all time steps. We expected that `MultiLayerNetwork.score` would compute the same value since our loss function is MSE, but it's intentionally computing a sum instead of the mean for time series data? 

That seems like it could be problematic, or at least confusing. For example, suppose I have time series with 1001 time steps and a truncated BPTT (forward and backward) length of 100. If I monitor the score during learning, won't I see a sudden (apparent) drop at the end of each epoch just because the last BPTT iteration has only 1 time step to pass over? If I have multiple time series with different lengths in my minibatch, then the last iteration might not be the only one affected this way. In general, it becomes impossible to compare the score in one BPTT iteration to the score in another. And might this have ramifications beyond monitoring?

**Comment by agibsonccc** [comment]
This has been fixed I believe?
